packages/blade/src/components/Tabs/Tabs.native.tsx

```diff
@@ -0,0 +1,214 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/no-non-null-asserted-optional-chain */
+/* eslint-disable babel/new-cap */
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+import React from 'react';
+import { TabBar, TabView } from 'react-native-tab-view';
+import { Dimensions } from 'react-native';
+import type { TabsProps } from './types';
+import { TabsContext } from './TabsContext';
+import { StyledTabButton } from './TabItem.native';
+import { textColor } from './tabTokens';
+import { iconSizeMap, useTabsItemPropRestriction } from './utils';
+import { TabIndicator } from './TabIndicator';
+import { SafeSceneMap } from './SafeSceneMap.native';
+import { getComponentId } from '~utils/isValidAllowedChildren';
+import { Text } from '~components/Typography';
+import { Box } from '~components/Box';
+import { useTheme } from '~utils';
+import { useControllableState } from '~utils/useControllable';
+import { useFirstRender } from '~utils/useFirstRender';
+
+const initialLayout = {
+  height: 0,
+  width: Dimensions.get('window').width,
+};
+
+const getTabs = (children: React.ReactNode): React.ReactElement[] => {
+  const childs = React.Children.toArray(children);
+  const tabList = childs.find((child) => getComponentId(child) === 'TabList');
+  if (!tabList || !React.isValidElement(tabList)) throw new Error('TabList is required');
+
+  return tabList.props.children;
+};
+
+const getTabPanels = (children: React.ReactNode): Record<string, () => React.ReactElement> => {
+  const childs = React.Children.toArray(children) as React.ReactElement[];
+  return childs
+    .filter((child) => getComponentId(child) === 'TabPanel')
+    .reduce((prev, curr) => {
+      return { ...prev, [curr.props.value]: () => curr.props.children };
+    }, {});
+};
+
+const getRoutes = (tabs: React.ReactElement[]) => {
```

Comment:
What is routes?

Reply:
It's an array of route object which TabView component expects. They basically map to the TabItems. 

https://reactnavigation.org/docs/tab-view/#tabview-props